### DISCUSSION TITLE
Restic restore changes original file permissions

### DISCUSSION THREAD

**Discussion by DanielXiao** [vmware-tanzu/velero#3609] [Community support Q&A] [answered]
I tried to migrate Harbor to a new cluster with Velero 1.5.3 + Restic, but its Postgres Pod did not start properly after restore. The error logs said: `FATAL: data directory “/var/lib/postgresql/data” has group or world access`, because the permission of `/var/lib/postgresql/data` was set to 755 after restore. It was 700 in the original persistent volume.
Is it an issue or expected behavior? I can find people are hitting the same error for backup/restore in the same cluster, see https://brianragazzi.wordpress.com/2020/08/14/use-helm-to-deploy-harbor-with-annotations-for-velero/.

**Comment by ashish-amarnath** [comment, marked as answer]
@DanielXiao I tried to repro this but wasn't successful. Here's what I tried..
![Uploading file-perms.gif…]()

**Reply by DanielXiao** [reply]
I can not see the screenshot, could you reupload it? I can reproduce this issue stably. Maybe you can delete the PV then try to restore it.

**Reply by ashish-amarnath** [reply]
![file-perms-1-fast](https://user-images.githubusercontent.com/22872138/112087713-b4610f80-8b4b-11eb-8fa2-54dffd6cd0a1.gif)

I realized that the earlier attachment was too big. But here you go.

**Reply by ashish-amarnath** [reply]
I am using restic to take volume backups, in case it wasn't clear.

**Reply by DanielXiao** [reply]
We use different storage classes. I did the test on vSphere CSI storage class, are there any different restore behaviors on that?

**Reply by ashish-amarnath** [reply]
I am not aware of such differences. But, even if there is any difference, then that won't be coming from Velero.
Are you able to attempt repro with a different storage class?

**Reply by ashish-amarnath** [reply]
I am going to mark this as answered. Please feel free to reach out if you need more help.

**Reply by DanielXiao** [reply]
Thank you
